mmcls/models/backbones/vision_transformer.py

```diff
@@ -374,101 +321,147 @@ def forward(self, x):
-            Default 2.
+            Default: 2
+        init_cfg (dict, optional): Initialization config dict
     """
+    arch_zoo = {
+        **dict.fromkeys(
+            ['s', 'small'], {
+                'embed_dims': 768,
+                'num_layers': 8,
+                'num_heads': 8,
+                'feedforward_channels': 768 * 3,
+                'qkv_bias': False
+            }),
+        **dict.fromkeys(
+            ['b', 'base'], {
+                'embed_dims': 768,
+                'num_layers': 12,
+                'num_heads': 12,
+                'feedforward_channels': 3072
+            }),
+        **dict.fromkeys(
+            ['l', 'large'], {
+                'embed_dims': 1024,
+                'num_layers': 24,
+                'num_heads': 16,
+                'feedforward_channels': 4096
+            }),
+    }
 
     def __init__(self,
-                 num_layers=12,
-                 embed_dim=768,
-                 num_heads=12,
+                 arch='b',
                  img_size=224,
                  patch_size=16,
                  in_channels=3,
-                 feedforward_channels=3072,
                  drop_rate=0.,
                  attn_drop_rate=0.,
+                 drop_path_rate=0.,
                  hybrid_backbone=None,
                  norm_cfg=dict(type='LN'),
                  act_cfg=dict(type='GELU'),
-                 num_fcs=2):
-        super(VisionTransformer, self).__init__()
-        self.embed_dim = embed_dim
+                 num_fcs=2,
+                 init_cfg=None):
+        super(VisionTransformer, self).__init__(init_cfg)
+
+        arch = arch.lower()
+        if isinstance(arch, str):
+            assert arch in set(self.arch_zoo), \
+                f'Arch {arch} is not in default archs {set(self.arch_zoo)}'
+            self.arch_settings = self.arch_zoo[arch]
+        else:
+            essential_keys = {
+                'embed_dims', 'num_layers', 'num_heads', 'feedforward_channels'
+            }
+            assert isinstance(arch, dict) and set(arch) == essential_keys, \
+                f'Custom arch needs a dict with keys {essential_keys}'
+            self.arch_settings = arch
+
+        self.embed_dims = self.arch_settings['embed_dims']
 
         if hybrid_backbone is not None:
             self.patch_embed = HybridEmbed(
                 hybrid_backbone,
                 img_size=img_size,
                 in_channels=in_channels,
-                embed_dim=embed_dim)
+                embed_dims=self.embed_dims)
         else:
-            self.patch_embed = PatchEmbed(
+            self.patch_embed = PatchEmbed_v2(
                 img_size=img_size,
-                patch_size=patch_size,
                 in_channels=in_channels,
-                embed_dim=embed_dim)
+                embed_dims=self.embed_dims,
+                kernal_size=patch_size,
+                stride=patch_size)
```

Comment:
We may move the args to `conv_cfg` (`patch_embed_cfg`) and it is the same in `HybridEmbed`.